### ISSUE TITLE
- Make the selection box searchable

### ISSUE THREAD

**Issue by ashrafalzyoud** [sk-ys/redmine_enhanced_ux#51]
If u can added this functionality for your plugin

**Comment by sk-ys** [comment]
I suggest that you install [Redmine UI extension](https://github.com/redmica/redmica_ui_extension) plugin or [Redmine searchable selectbox](https://github.com/farend/redmine_searchable_selectbox) plugin.

**Comment by ashrafalzyoud** [comment]
im use [Redmine UI extension](https://github.com/redmica/redmica_ui_extension)
plz im send u email in [yskrmt@gmail.com](mailto:yskrmt@gmail.com)  plz check

**Comment by ashrafalzyoud** [comment]
Mr @sk-ys 
Any update about that conflict the page still jump when I select any field have select2 value

**Comment by sk-ys** [comment]
Hi, Mr. @ashrafalzyoud .
I apologize for the long silence. I have reproduced this issue in my environment recently.
I checked it with the `RedMica UI extension plugin` and the `Redmine Rags plugin` by RedmineUP.
https://github.com/user-attachments/assets/6f580fb9-50f8-412d-aaca-3731d7cbde18

This phenomenon occurs due to the following line of code in `RedMica UI extension plugin`:
```javascript
$(document).on('select2:open', function() {
  document.querySelector('.select2-search__field').focus();
});
```
https://github.com/redmica/redmica_ui_extension/blob/32589e603ea000a5889244a24d6810d6f2bd75f2/assets/javascripts/searchable_selectbox/searchable_selectbox.js#L15-L19

This code is intended to focus on the element with the `select2-search__field` class when Select2 is opened.
However, the `select2-search__field` class is used by multiple plugins, resulting in multiple elements on the page.
Consequently, `document.querySelector('.select2-search__field')` returns the first matching element in the document.
As a result, if the first matching element belongs to Select2 from the Redmine Tags Plugin, unexpected scrolling occurs.
![Image](https://github.com/user-attachments/assets/9bb81f3e-4630-4baa-8bd3-2d2949bfa07d)

In conclusion, I suspect that the cause of this issue lies in the `RedMica UI extension plugin`.

**Comment by sk-ys** [comment]
@ashrafalzyoud 
I have created a customized version of the RedMica UI extension that supports Redmine 5.
You can find it here: https://github.com/sk-ys/redmica_ui_extension/tree/redmine5

To install it, use the following command:
```
git clone https://github.com/sk-ys/redmica_ui_extension.git -b redmine5
```
Note: You need to uninstall the original version of the RedMica UI extension plugin before installing it.
